### PR TITLE
feat: add Qwik JSX compiler

### PR DESCRIPTION
**Pull request by userquin**
This PR just adds a new option to `jsx` to allow use it with Qwik, no idea if the types can be shared, the example just works.

The idea is to have another "compiler" for Qwik.

PR in draft until some Qwik expert review this:

imagen

Check this SB repro with custom compiler using `@svgr/core` + `@svgr/plugin-jsx` plugin (works in SB while `@svgx/core` doesn't):
https://stackblitz.com/edit/unplugin-unplugin-icons-p2lwmh

### CODE REVIEW

File: examples/vite-qwik/vite.config.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ export default defineConfig(() => {
       Icons({
         autoInstall: true,
         compiler: 'qwik',
+        // compiler: 'jsx',
```
Comment (Aslemammad) [inline examples/vite-qwik/vite.config.ts:12]: Do we need these comments? what about the jsx flag?

**Comment by Aslemammad** [comment]
Maybe Jack can help also! He's an active developer in the qwik community. cc @thejackshelton 

Also my mates, @AmirSa12 @araqaee can share what they know.

File: examples/vite-qwik/vite.config.ts (hunk 1 above)
Comment (userquin) [reply examples/vite-qwik/vite.config.ts:12]: We don't need qwik compiler, it is already there, the problem is using qkik compiler in StackBlitz, it is not working.

This PR just adds another way to use unplugins icons in qwik.

Comment (userquin) [reply examples/vite-qwik/vite.config.ts:12]: check this another PR, links to SB on examples folder: https://github.com/unplugin/unplugin-icons/pull/315

Comment (userquin) [reply examples/vite-qwik/vite.config.ts:12]: change it in SB

Comment (userquin) [reply examples/vite-qwik/vite.config.ts:12]: updating SB script and the vite.config.ts file to keep only the `qwik` compiler

**Comment by userquin** [comment]
You can check it with this link: https://stackblitz.com/fork/github/unplugin/unplugin-icons/tree/userquin/feat-add-qwik-jsx-compiler/examples/vite-qwik

Since we don't have yet the new option in `jsx` TS complains in `vite.config.ts`...

**Review by Aslemammad — APPROVED** [review]

**Review by antfu — APPROVED** [review]

**Pull request merged**